### PR TITLE
NAC-143: picture renditions generation

### PR DESCRIPTION
**Pull request by Nishant0928**

### CODE REVIEW

**Review by rakeshkumar1019 — APPROVED** [review]

File: nuxeo-admin-console-web/angular-app/src/app/features/sub-features/generic-multi-feature-layout/generic-multi-feature-layout.mapping.ts [hunk 1]
```diff
@@ -94,4 +96,55 @@ export const featureMap = () => ({
       data,
     };
   },
+  [FEATURES.PICTURES]: (tabType: string) => {
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/features/sub-features/generic-multi-feature-layout/generic-multi-feature-layout.mapping.ts:99]: PICTURE_RENDITIONS

**Review by madhurkulshrestha-hyland — CHANGES_REQUESTED** [review]

File: nuxeo-admin-console-web/angular-app/src/app/shared/constants/rest-end-ponts.constants.ts [hunk 2]
```diff
@@ -48,5 +49,9 @@ export const REST_END_POINT_CONFIG: Record<RestEndpointKey, RestEndpointConfig>
     THUMBNAIL_GENERATION: {
         endpoint: "/management/thumbnails/recompute",
         method: "POST"
+    },
+    PICTURES: {
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/shared/constants/rest-end-ponts.constants.ts:53]: PICTURE_RENDITIONS

File: nuxeo-admin-console-web/angular-app/src/app/layouts/menu-bar/menu-bar.constants.ts [hunk 3]
```diff
@@ -39,4 +45,5 @@ export const ROUTES_TITLE = {
   BULKACTIONMONITORING: "Bulk Action Monitoring",
   PROBES: "Probes",
   THUMBNAIL_GENERATION: "Thumbnail Generation",
+  PICTURES: "PICTURES",
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/layouts/menu-bar/menu-bar.constants.ts:48]: PICTURE_RENDITIONS

File: nuxeo-admin-console-web/angular-app/src/app/shared/constants/rest-end-ponts.constants.ts [hunk 4]
```diff
@@ -6,7 +6,8 @@ export const REST_END_POINTS = {
     CAPABILITIES: "CAPABILITIES",
     LOGOUT: "LOGOUT",
     BULK_ACTION_MONITORING: "BULK_ACTION_MONITORING",
-    THUMBNAIL_GENERATION: "THUMBNAIL_GENERATION"
+    THUMBNAIL_GENERATION: "THUMBNAIL_GENERATION",
+    PICTURES: "PICTURES"
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/shared/constants/rest-end-ponts.constants.ts:10]: PICTURE_RENDITIONS

**Review by madhurkulshrestha-hyland — CHANGES_REQUESTED** [review]

File: nuxeo-admin-console-web/angular-app/src/app/layouts/menu-bar/menu-bar.constants.ts (hunk 3 above)
Comment (Nishant0928) [reply nuxeo-admin-console-web/angular-app/src/app/layouts/menu-bar/menu-bar.constants.ts]: done

File: nuxeo-admin-console-web/angular-app/src/app/features/pictures/pictures-renditions.constants.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    RENDITIONS_BUTTON_LABEL: "GENERATE",
+    DOCUMENT_QUERY:
+    "ecm:path='{query}' AND ecm:mixinType = 'Picture' AND ecm:isProxy = 0 AND ecm:mixinType != 'HiddenInNavigation' AND ecm:isVersion = 0 AND ecm:isTrashed = 0",
+  FOLDER_QUERY:
+    "ecm:ancestorId='{query}' AND ecm:mixinType = 'Picture' AND ecm:isProxy = 0 AND ecm:isVersion = 0",
+  NXQL_QUERY:
+    "ecm:uuid='{query}' OR ecm:ancestorId='{query}' AND ecm:mixinType = 'Picture' AND ecm:isProxy = 0 AND ecm:isVersion = 0",
+  NXQL_QUERY_DEFAULT_VALUE:
+    "SELECT * FROM Document WHERE ecm:mixinType = 'Picture' AND picture:views/*/title IS NULL AND ecm:isVersion = 0 AND ecm:isProxy = 0 AND ecm:isTrashed = 0 AND dc:title = 'A picture without renditions'",
+  PICTURES: "Pictures",
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/features/pictures/pictures-renditions.constants.ts:14]: Is this constant required ?

Comment (Nishant0928) [reply nuxeo-admin-console-web/angular-app/src/app/features/pictures/pictures-renditions.constants.ts]: not required as of now , removed it

**Review by madhurkulshrestha-hyland — APPROVED** [review]

**Pull request merged**